From df8451015333a4f7d84b386d3f252aa787413e46 Mon Sep 17 00:00:00 2001
From: John Linville <linville@tuxdriver.com>
Date: Tue, 23 Jun 2015 15:04:03 -0400
Subject: [PATCH] mwifiex: do not short circuit exit from mwifiex_set_mgmt_ies

Without this change, the code simply exits after calling
mwifiex_uap_set_head_tail_ies, leving the call to
mwifiex_set_mgmt_beacon_data_ies as dead code.

Coverity CID #1271292

Signed-off-by: John W. Linville <linville@tuxdriver.com>
Acked-by: Amitkumar Karwar <akarwar@marvell.com>
Signed-off-by: Kalle Valo <kvalo@codeaurora.org>

* Fixes broken WAC
* cherry picked from 4.3 kernel 8b2c621c9f6f6240edaf4359c3148068fff8b1c7

Signed-off-by: Radek Dostal <radek.dostal@streamunlimited.com>
---
 drivers/net/wireless/mwifiex/ie.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/wireless/mwifiex/ie.c b/drivers/net/wireless/mwifiex/ie.c
index 7371541..30b13d9 100644
--- a/drivers/net/wireless/mwifiex/ie.c
+++ b/drivers/net/wireless/mwifiex/ie.c
@@ -404,6 +404,8 @@ int mwifiex_set_mgmt_ies(struct mwifiex_private *priv,
 	int ret;
 
 	ret = mwifiex_uap_set_head_tail_ies(priv, info);
+
+	if (ret)
 		return ret;
 
 	return mwifiex_set_mgmt_beacon_data_ies(priv, info);
-- 
1.9.1

